src/TLuaInterpreter.cpp

```diff
@@ -702,35 +700,34 @@ int TLuaInterpreter::selectString(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#selectCurrentLine
 int TLuaInterpreter::selectCurrentLine(lua_State* L)
 {
-    std::string luaSendText = "";
-    if (lua_gettop(L) == 0) {
-        luaSendText = "main";
-    } else {
+    std::string windowName;
```

Comment:
That's another fun design issue. Right now the calls to find the correct subconsole are nicely spread between mudlet.cpp, TConsole.cpp, and one call in TLuaInterpreter. That's a lot of code, all of which does the exact same thing.

Why do we need that?

I'd much rather have a single `getNamedConsole(windowName)` method in TConsole that returns `this` if the name indicates the main console, `null` if it can't be found, and the named sub-`TConsole` instance otherwise. That'd allow us to remove all `windowName` parameters, as well as all sub-console-dispatch functions from mudlet.cpp.

Reply:
Actually the main `TConsole` is actually the entry-point into access into the profile's other `TConsole` or at least the ones that the user creates and interacts with (Oh! ***And*** the `TLabel`s).

At the moment, though, there are an unhelpfully large number of functions (probably dating back to a time when Mudlet could only open one profile at a time ???) that instead go via the `mudlet` singleton class (specifying a `Host` AND a console name) that end up being routed to the Main `TConsole` anyhow. {The reason being that that console marshals the others because it holds the list of pointers to the other ones - and because some of the other - the sub-consoles - are contained inside it and when it is destroyed they must be as well.}

The bottom line is that we should be aiming to go to the main `TConsole` (which *should* always be present) for all console related functions and it should either do what is asked for, either to itself, or to one of the other class instances that it controls, and return either the wanted data - if appropriate or a success or failure indication, and if there is more than one failure cause it should indicate which.